### PR TITLE
[Bitcoin] Redeem utxos locked by uncompressed public key hash

### LINKED ISSUE #1432: [Bitcoin] Support uncompressed public key in TransactionSigner::keyForPublicKeyHash
Check `TWPublicKeyTypeSECP256k1` and `TWPublicKeyTypeSECP256k1Extended`

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #1432

### CODE REVIEW

File: src/Bitcoin/TransactionSigner.cpp [hunk 1]
```diff
@@ -273,9 +273,11 @@ Data TransactionSigner<Transaction, TransactionBuilder>::pushAll(const std::vect
...
-        auto publicKey = PrivateKey(key).getPublicKey(TWPublicKeyTypeSECP256k1);
-        auto keyHash = TW::Hash::ripemd(TW::Hash::sha256(publicKey.bytes));
-        if (keyHash == hash) {
+        auto pubKeyExtended = PrivateKey(key).getPublicKey(TWPublicKeyTypeSECP256k1Extended);
+        auto pubKey = pubKeyExtended.compressed();
+        auto keyHash = Hash::sha256ripemd(pubKey.bytes.data(), pubKey.bytes.size());
+        auto keyHash2 = Hash::sha256ripemd(pubKeyExtended.bytes.data(), pubKeyExtended.bytes.size());
+        if (keyHash == hash || keyHash2 == hash) {
             return Data(key.begin(), key.end());
         }
```
Comment (catenocrypt) [inline src/Bitcoin/TransactionSigner.cpp:282]:
```suggestion
        auto keyHash = Hash::sha256ripemd(pubKey.bytes.data(), pubKey.bytes.size());
        auto keyHashExt = Hash::sha256ripemd(pubKeyExtended.bytes.data(), pubKeyExtended.bytes.size());
        if (keyHash == hash || keyHashExt == hash) {
            return Data(key.begin(), key.end());
        }
```

File: src/Bitcoin/TransactionSigner.cpp [hunk 2]
```diff
@@ -273,9 +273,11 @@ Data TransactionSigner<Transaction, TransactionBuilder>::pushAll(const std::vect
 template <typename Transaction, typename TransactionBuilder>
 Data TransactionSigner<Transaction, TransactionBuilder>::keyForPublicKeyHash(const Data& hash) const {
     for (auto& key : input.private_key()) {
-        auto publicKey = PrivateKey(key).getPublicKey(TWPublicKeyTypeSECP256k1);
-        auto keyHash = TW::Hash::ripemd(TW::Hash::sha256(publicKey.bytes));
-        if (keyHash == hash) {
+        auto pubKeyExtended = PrivateKey(key).getPublicKey(TWPublicKeyTypeSECP256k1Extended);
+        auto pubKey = pubKeyExtended.compressed();
+        auto keyHash = Hash::sha256ripemd(pubKey.bytes.data(), pubKey.bytes.size());
+        auto keyHash2 = Hash::sha256ripemd(pubKeyExtended.bytes.data(), pubKeyExtended.bytes.size());
```
Comment (catenocrypt) [inline src/Bitcoin/TransactionSigner.cpp:279]: Extended keyhash computation can be skipped if non-extended keyHash matches (minor optimization)

File: swift/Tests/Blockchains/BitcoinTests.swift [hunk 3]
```diff
@@ -127,4 +127,53 @@ class BitcoinTransactionSignerTests: XCTestCase {
...
+        // genre awake harsh charge accident census click turn parade solution drill frown
+        // history allow jelly tenant hollow task disagree develop hood nature raise theory
+
+        // compressed WIF
+        let wif = "L4BeKzm3AHDUMkxLRVKTSVxkp6Hz9FcMQPh18YCKU1uioXfovzwP"
+        let decoded = Base58.decode(string: wif)!
+        let key = PrivateKey(data: decoded[1 ..< 33])!
+        let pubkey = key.getPublicKeySecp256k1(compressed: false)
+
+        // shortcut methods only support s compressed public key
```
Comment (catenocrypt) [inline swift/Tests/Blockchains/BitcoinTests.swift:141]:
```suggestion
        // shortcut methods only support compressed public key
```

**Review by catenocrypt — APPROVED** [review]

File: src/Bitcoin/TransactionSigner.cpp (hunk 2 above)
Comment (hewigovens) [reply src/Bitcoin/TransactionSigner.cpp]: should be fixed

**Review by vikmeup — APPROVED** [review]

**Comment by hewigovens** [comment]
Don't merge this yet, going to verify with real tx

**Pull request merged**